### PR TITLE
ICU-21900 integrate CLDR release-41-alpha1 to ICU main for 71 front-load

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21900
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

This PR integrates CLDR release-41-alpha1 (not yet public alpha) into the ICU main branch as part of the ICU 71 front-loading work. There are some things to call out:

1. This attempts to generate the new Japanese extensions data in icu4c/source/data/brkitr/ja.txt from xml data in icu4c/source/data/xml/brkitr/ja.xml, using an updated ldmlICU.dtd and enhancements to tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/mapper/BreakIteratorMapper.java. This is not yet fully working (only one of the extension entries is generated), so instead the data in icu4c/source/data/brkitr/ja.txt is just recopied from the current committed version. This will be completed in a followup PR.
2. Until now, certain CLDR dtds have been copied into versioned subdirectories in icu4c/source/data/dtd. It is not clear that we need the versioned directories, so this PR starts a new unversioned directory with the latest CLDR dtds.
3. CLDR 41 alpha1 removes some bad incorrect-order list patterns in ml and ur_IN. There were some tests specific to these, not clear we need those tests anymore (here they are just updated).
4. This PR adds one more locale, ks_Deva

### CODE REVIEW

**Comment by markusicu** [comment]
> 1. This attempts to generate the new Japanese extensions data

David replied in email:
> There's a mapping of paths to output files you can change.

https://github.com/unicode-org/icu/blob/main/tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/LdmlConverter.java#L661

This isn't in the build script of config since it's not something we change often. I hope it's self-evident what's going on, but I'd be totally happy if you wanted to add more docs explaining it better.

> 2. Until now, certain CLDR dads have been copied into versions subdirectories in icu4c/source/data/dtd. It is not clear that we need the versioned directories, so this PR starts a new unversioned directory with the latest CLDR dtds.

“CLDR dads”? Moms? dtds?

**Comment by pedberg-icu** [comment]
> > There's a mapping of paths to output files you can change.
> 
> https://github.com/unicode-org/icu/blob/main/tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/LdmlConverter.java#L661

Made that change which fixed one problem with the extensions generation, still one remaining, for the next PR.

> > 2. Until now, certain CLDR dads have been copied into versions subdirectories in icu4c/source/data/dtd. It is not clear that we need the versioned directories, so this PR starts a new unversioned directory with the latest CLDR dtds.
> 
> “CLDR dads”? Moms? dtds?

Fixed that autocorrect typo in PR description.

File: icu4c/source/test/intltest/transtst.cpp [hunk 1]
```diff
@@ -3654,19 +3654,35 @@ void TransliteratorTest::TestIncrementalProgress(void) {
...
+                    // 7. Any-Braille/*
+                    // 8. Any-*/Gurage_2013
+                    // 9. Any-*/Gutgarts
+                    // 10. Any-*/Tekie_Alibekit
+                    // 11. Any-*/Xaleget
+                    //
                     if (    id.compare((UnicodeString)"Devanagari-Arabic/") != 0
                          && !(id.startsWith((UnicodeString)"Any-") &&
-                                (id.endsWith((UnicodeString)"/BGN") || id.endsWith((UnicodeString)"/BGN_1981") || id.endsWith((UnicodeString)"/UNGEGN") || id.endsWith((UnicodeString)"/MNS"))
+                                (id.endsWith((UnicodeString)"/BGN") || id.endsWith((UnicodeString)"/BGN_1981") || id.endsWith((UnicodeString)"/MNS"))
```
Comment (markusicu) [inline icu4c/source/test/intltest/transtst.cpp:3675]: FYI
Instead of `(UnicodeString)"/BGN"` you should be able to write `u"/BGN"`.

**Review by markusicu — APPROVED** [review]
lgtm

File: icu4c/source/test/intltest/transtst.cpp (hunk 1 above)
Comment (pedberg-icu) [reply icu4c/source/test/intltest/transtst.cpp:3675]: Thanks, was just following the existing style. But I can update all of usages here for the alpha2 integration.

**Pull request merged**